application/models/message_test.go

```diff
@@ -195,6 +196,82 @@ func (ms *ModelSuite) TestMessage_Create() {
 	}
 }
 
+func (ms *ModelSuite) TestMessage_CreateForRest() {
+	t := ms.T()
+
+	f := Fixtures_Message_Create(ms, t)
+	threadUUID := f.Threads[0].UUID.String()
+
+	tests := []struct {
+		name        string
+		user        User
+		requestUUID string
+		threadUUID  *string
+		content     string
+		wantErr     bool
+	}{
+		{
+			name:        "good, 1st message, visible request",
+			user:        f.Users[2],
+			requestUUID: f.Requests[1].UUID.String(),
+			threadUUID:  nil,
+			content:     "Owe nothing to anyone, except to love one another.",
+			wantErr:     false,
+		},
+		{
+			name:        "good, already a participant",
+			user:        f.Users[0],
+			requestUUID: f.Requests[1].UUID.String(),
+			threadUUID:  &threadUUID,
+			content:     "Hatred stirs up conflict, but love covers over all wrongs.",
+			wantErr:     false,
+		},
+		{
+			name:        "bad, not a participant",
+			user:        f.Users[1],
+			requestUUID: f.Requests[1].UUID.String(),
+			threadUUID:  &threadUUID,
+			content:     "bad message",
+			wantErr:     true,
+		},
+		{
+			name:        "bad, not a visible request",
+			user:        f.Users[1],
+			requestUUID: f.Requests[0].UUID.String(),
+			threadUUID:  nil,
+			content:     "another bad message",
+			wantErr:     true,
+		},
+		{
+			name:        "bad, thread not valid for request",
+			user:        f.Users[0],
+			requestUUID: f.Requests[2].UUID.String(),
+			threadUUID:  &threadUUID,
+			content:     "another bad message",
```

Comment:
Making the content distinct, e.g. "yet another bad message" might be helpful for quicker debugging/isolating which case fails, though the distinct names are almost certainly sufficient so it's probably not worth changing.

Reply:
:+1: